### DISCUSSION TITLE
Briefly on processes

### DISCUSSION THREAD

**Discussion by iminavo** [iminavo/scrum#14] [Announcements]
TLDR;
fewer ceremonies, more freedom

Our current processes are more relaxed. We used to have more ceremonies in terms of project management, e.g. pointing tasks in a way to generate an _acceptable_ burndown chart, and now we use the points as a tool for limiting the amount of work per developer per sprint. I find the latter a more appropriate use of points. We also have a _simpler_ system of pointing tasks, 5 means the whole week. I'm not a big fan of binding points to time but I can live with this since it does not do much damage.

We have shorter sprints, 1 week as opposed to 2 weeks before. I think this is a good thing since it makes the team more responsive to changes in the requirements and within the team itself. We are not rolling out features at the end of each sprint though, probably because we still split the work artificially, so no big improvements there. We still have stories that are arbitrary in some sense and we are still doing a variation of a waterfall, at least in the part of the system I am currently working on. There is this sprint goal idea, which is ok and may project measurable commitment, but it does not make sense if the tasks for a sprint are _contrived_. I and a few fellow engineers tried to push the idea of working closely with the client to formulate the stories, but this was pushed back mainly by Dave, so we took on his approach of building big chunks of the system and then connecting them /vertically/, instead of growing the system use-case by use-case /horizontally/.

We are also fine with carrying over tasks from one sprint to another, which is a sign of preferring reality over the _plan_. 

The team became more self-organized, we are not shy to organize calls when we need to discuss things, and we try to take matters into our own hands, whenever possible I'd say I'm happy with the results. We still lack a few important freedoms over our codebase /e.g. the merge button and the pipelines are not in our hands/, but overall we are in charge of the engineering practices. The merge button is still not in our hands though, so the supposedly short-living branches live for several days and PRs stack one on top of the other.

Overall, I'd say we are moving towards better processes.

**Comment by f-person** [comment]
> The merge button is still not in our hands though

Wow, how's that possible? Who's merging patches, then?

**Reply by iminavo** [reply]
Yeah, right? :)))

Phillip does. This is a trust issue. Previously the code produced by us was full of defects and at some point, the client chose to police all the PRs. Phillip himself does not like this decision but his boss requires him to do so. We have to deal with this until we earn the client's trust, which is /only?/ possible by releasing features that will actually solve problems and be stable over time, which we cannot do quickly since we chose to build the system not per use-case but per block\service, so we have to wait for some _foundation_ to be ready. 

Bear in mind that once the pieces of the foundation are ready, they, most probably, won't integrate easily. Plus, and this happened to us last week, the third-party API we were integrating got an update so the foundation needs to be rethought, which also will set us back time-wise. This happens and is more tolerable when affecting values /actually useful code/ then waste /potentially useful code/. But this is a different, yet very interesting, topic.

**Reply by f-person** [reply]
Wow, that sounds somewhat weird! There's a clever saying, "If you hire smart people, listen to them". I guess Steve Jobs was the one who popularized this idea. I understand the client's point of view, too, though. However, it seems to be based only on intuition which isn't the best strategy, in my opinion — anyways, wishing you luck with improving the processes and returning the trust!

**Reply by iminavo** [reply]
Thanks!

If you are interested in these topics I suggest you read or skim through the [Extreme Programming Explained
](https://www.oreilly.com/library/view/extreme-programming-explained/0201616416/) by Kent Beck and his latest _Software Design: Tidy First?_ ideas. [Here's](https://tidyfirst.substack.com/feed) the RSS feed for the latter.
